### PR TITLE
[ENH] Introducing inverse_transform into StandardScaler

### PR DESCRIPTION
**Pull request by dguijo**
#### What does this implement/fix? Explain your changes.

Includes the possibility to perform the inverse transform to obtain the original representation. 

#### Does your contribution introduce a new dependency? If yes, which one?

Nope

### CODE REVIEW

**Review by hadifawaz1999 — COMMENTED** [review]
Thanks @dguijo ! looks good but could you maybe change the existing example while at the same time continue the example to use the inverse transform ? for instance the example used for now is odd, input 0s and output 0s, maybe better to use a real example ?

File: aeon/transformations/collection/scaler.py [hunk 1]
```diff
@@ -74,35 +79,75 @@ class TimeSeriesScaler(BaseCollectionTransformer):
...
+    def __init__(
+        self, copy=True, with_mean=True, with_std=True, inverse_transform_needed=False
+    ):
         self.with_mean = with_mean
         self.with_std = with_std
         self.copy = copy
         self.scaler = StandardScaler(copy=copy, with_mean=with_mean, with_std=with_std)
+        self.inverse_transform_needed = inverse_transform_needed
+        if self.inverse_transform_needed:
+            self._tags["capability:inverse_transform"] = True
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/collection/scaler.py:91]: Could you add a check for this in the test? I don't think this is setting correctly to be honest. See https://github.com/aeon-toolkit/aeon/blob/main/aeon/classification/compose/_pipeline.py for a (more complicated) version of dynamic tags

**Review by MatthewMiddlehurst — COMMENTED** [review]
Given that you must first transform the data to be able to do the inverse transform, is there a case where this will be useful (as you already have the data).

Also I'm surprised this didnt break the testing, I thought transform was supposed to keep the same state. Saving the objects is better suited to fit.

**Comment by dguijo** [comment]
Hello, Matt:
There is some point in having the inverse transform, but maybe not following the way this function is designed ATM. 

Imagine that you have created new synthetic patterns from the training set when it was already standarised and you want to get the original representation of these synthetic patterns to do some plotting or explainability, or whatever. For this,  you need to have the inverse transform. 

However, this can't be achieved now as this transformer is being performed ATM, as it fits a StandardScaler per time series (n_channels x n_timepoints) instead of doing it per channel (n_timeseries x n_timepoints). 

It can be used in time series reconstruction (get the original representation of the reconstructed time series), time series averaging (get the original representation of the averaged time series), and so on. I must admit it is not a typical behaviour in standard machine learning tasks, but I find it useful :smile:.

So, for me, I think we have to give a thought to the current design, and think if the other one (one StandardScaler per channel) is equally useful and then: 1) have both of them, 2) decide which to have.

Thanks!

**Comment by MatthewMiddlehurst** [comment]
I think to do this properly you would have to introduce a `_fit` method and change the `fit_is_empy` tag (perhaps dynamically like you are with inverse transform?).

**Comment by TonyBagnall** [comment]
@dguijo closing this if its ok, it will now be covered here https://github.com/aeon-toolkit/aeon/issues/2075

**Pull request closed without merging**